addons/purchase/tests/test_purchase.py

```diff
@@ -942,3 +942,54 @@ def test_vendor_price_by_purchase_order_company(self):
         self.assertEqual(po.amount_untaxed, 15.0)
         po.company_id = company_a.id
         self.assertEqual(po.amount_untaxed, 10.0)
+
+    def test_bill_in_purchase_matching_individual(self):
+        """
+        Tests that if the vendor is an individual with a company, the bill will still appear when
+        using the purchase matching button on a vendor bill
+        """
+        company_partner = self.env['res.partner'].create({
+            'name': 'Small Company',
+            'company_type': 'company',
+        })
+        individual_partner = self.env['res.partner'].create({
+            'name': 'Big Vendor',
+            'company_type': 'person',
+            'parent_id': company_partner.id,
+        })
+        downpayment_product = self.env['product.product'].create({
+            'name': 'Downpayment Service',
+            'type': 'service',
+            'invoice_policy': 'order',
+        })
+        purchase_order = self.env['purchase.order'].create({
+            'partner_id': individual_partner.id,
+            'company_id': self.env.company.id,
+            'order_line': [(0, 0, {
+                'product_id': downpayment_product.id,
+                'name': 'Downpayment Service',
+                'product_qty': 1,
+                'price_unit': 100,
+                'date_planned': '2025-01-01',
+            })]
+        })
+        purchase_order.button_confirm()
+        vendor_bill = self.env['account.move'].create({
+            'move_type': 'in_invoice',
+            'partner_id': individual_partner.id,
+            'company_id': self.env.company.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Downpayment Service',
+                'product_id': downpayment_product.id,
+                'quantity': 1,
+                'price_unit': 100,
+                'account_id': self.env['account.account'].search([], limit=1).id,
+            })]
+        })
+        result = vendor_bill.action_purchase_matching()
+        matching_records = self.env['purchase.bill.line.match'].search(result['domain'])
+        matched_bill_ids = matching_records.mapped('account_move_id.id')
+        self.assertIn(vendor_bill.id, matched_bill_ids, "Vendor bill should be in the matching results")
```

Comment:
Two comments
\- Provide error message when it explains the expected value only. Here, the line is explicit enough. (It's like doing `self.assertEqual(price, 2, "The price should be $2")`: the error message doesn't provide anything. Most of the times, you can just leave it empty
\- It would be nice to be more complete, something like:
```py
self.assertEqual(len(matching_records), 2)
self.assertEqual(matching_records.account_move_id, vendor_bill)
self.assertEqual(matching_records.purchase_order_id, purchase_order)
```
(According to my attempt, you may need to flush before searching for the match lines)